### ISSUE TITLE
unsharing not possible when offline

### ISSUE THREAD

**Issue by JKawohl** [owncloud/client#5384]
### Expected behaviour
Unshare option should be possible when offline

### Actual behaviour

![screen shot 2016-12-14 at 17 20 40](https://cloud.githubusercontent.com/assets/12275313/21190504/4b11ec94-c222-11e6-88eb-d2bafae0c39f.png)

After the right click there is only a spinning wheel

### Steps to reproduce
1. Create share
2. go offline with your computer
3. try to remove share (right click........)

@hodyroff 

I think there could be two approaches:

1. Create verbose error message.
2. Store unshare locally and resync it when only again

**Comment by SamuAlfageme** [comment]
@Kawohl I'd propose a third option: Disable the `Share with owncloud` option the same way we do when the sync is paused. (And the way, it should be when a local file is yet to be uploaded to the server). 

The second option would be a really nice enhancement but I think it'll increase unnecessarily the complexity (we'll need to handle a queue of this kind of requests when back online, etc.).

Thanks!

**Comment by hodyroff** [comment]
In some ways we have to do this anyway as people can have done offline deletes, renames, moves, etc.!
Just yesterday one of the sales people decided to restructure shared folders in the train while beeing offline - because she had time ...
The result was full of corner cases.
@SamuAlfageme Basically go offline and start moving around shared folders and do all kinds of strange stuff. Delete stuff you have no delete rights for. Files. Move Folders around from one share to another, etc. Result was a hanging ownCloud client and 3 of 671 files which had to be somewhat taken care of manually, by removing them from the client folder, etc. please think about how this can be tested so we can identify the corner cases.

**Comment by SamuAlfageme** [comment]
@hodyroff yes, agreed: I am creating an issue for an offline/paused test plan for the desktop client in https://github.com/owncloud/QA/issues/338, and sometime soon we would need to continue thinking about the state machine underlying the client (#5321)
